app/scripts/migrations/051.js

```diff
@@ -0,0 +1,29 @@
+import { cloneDeep } from 'lodash'
+import { NETWORK_TYPE_TO_ID_MAP } from '../controllers/network/enums'
+
+const version = 51
+
+/**
+ * Set the chainId in the Network Controller provider data for all infura networks
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    versionedData.data = transformState(state)
+    return versionedData
+  },
+}
+
+function transformState(state) {
+  const type = state?.NetworkController?.provider?.type
+
+  if (NETWORK_TYPE_TO_ID_MAP[type]) {
```

Comment:
Should we also check if `provider.chainId` is already set here?

I don't think it matters, but, a thought anyway.

Reply:
sure, yeah